pkg/capabilities/consensus/ocr3/types/aggregator.go

```diff
@@ -34,6 +39,146 @@ func (m *Metadata) padWorkflowName() {
 	}
 }
 
+// Encode serializes Metadata in contract order:
+// 1B Version, 32B ExecutionID, 4B Timestamp, 4B DONID, 4B DONConfigVersion,
+// 32B WorkflowID, 10B WorkflowName, 20B WorkflowOwner, 2B ReportID
+func (m Metadata) Encode() ([]byte, error) {
+	m.padWorkflowName()
+	buf := new(bytes.Buffer)
+
+	// 1) Version as a single byte
+	if err := buf.WriteByte(byte(m.Version)); err != nil {
+		return nil, err
+	}
+
+	// 2) Helper to decode a hex string and ensure length
+	writeHex := func(field string, expectedBytes int) error {
+		s := strings.TrimPrefix(field, "0x")
+		b, err := hex.DecodeString(s)
+		if err != nil {
+			return fmt.Errorf("invalid hex in field: %w", err)
+		}
+		if len(b) != expectedBytes {
+			return fmt.Errorf("wrong length: expected %d bytes, got %d", expectedBytes, len(b))
+		}
+		_, err = buf.Write(b)
+		return err
+	}
+
+	// ExecutionID: 32 bytes
+	if err := writeHex(m.ExecutionID, 32); err != nil {
+		return nil, fmt.Errorf("ExecutionID: %w", err)
+	}
+
+	// Timestamp, DONID, DONConfigVersion—all 4‐byte big endian
+	for _, v := range []uint32{m.Timestamp, m.DONID, m.DONConfigVersion} {
+		if err := binary.Write(buf, binary.BigEndian, v); err != nil {
+			return nil, err
+		}
+	}
+
+	// WorkflowID: 32 bytes
+	if err := writeHex(m.WorkflowID, 32); err != nil {
+		return nil, fmt.Errorf("WorkflowID: %w", err)
+	}
+
+	// Workflow Name: 10 bytes
+	if err := writeHex(m.WorkflowName, 10); err != nil {
+		return nil, fmt.Errorf("WorkflowName: %w", err)
+	}
+
+	// WorkflowOwner: 20 bytes
+	if err := writeHex(m.WorkflowOwner, 20); err != nil {
+		return nil, fmt.Errorf("WorkflowOwner: %w", err)
+	}
+
+	// ReportID: 2 bytes
+	if err := writeHex(m.ReportID, 2); err != nil {
+		return nil, fmt.Errorf("ReportID: %w", err)
+	}
+
+	return buf.Bytes(), nil
+}
+
+const MetadataLen = 1 + 32 + 4 + 4 + 4 + 32 + 10 + 20 + 2 // =109
```

Comment:
Maybe worth repeating these details here :shrug: 
```suggestion
// 1B Version, 32B ExecutionID, 4B Timestamp, 4B DONID, 4B DONConfigVersion,
// 32B WorkflowID, 10B WorkflowName, 20B WorkflowOwner, 2B ReportID
const MetadataLen = 1 + 32 + 4 + 4 + 4 + 32 + 10 + 20 + 2 // =109
```